en/02_Developer_Guides/03_Forms/Field_types/03_HTMLEditorField.md

```diff
@@ -444,32 +408,148 @@ elements (and back out in case an existing element gets edited).
+
+## Developing a wrapper to use a different HTML editor with `HTMLEditorField`
+
+There are two part to adding a new HTML editor implementation to Silverstripe CMS - a `HTMLEditorConfig` subclass and a JavaScript implementation.
+
+### Adding a custom `HTMLEditorConfig`
+
+The [`HTMLEditorConfig`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig) class gives all the abstraction necessary to add your own HTML editor implementation to Silverstripe CMS. Make a new subclass of `HTMLEditorConfig` and implement all the abstract methods.
+
+You must define the [`protected string $schemaComponent`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig->schemaComponent) property, setting it to the name of the react component that will be used in the CMS.
+
+You must also define the [`protected string $configType`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig->configType) property. Set this to a unique string which will be used to identify this config class in the CMS JavaScript.
+
+In the [`init()`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig::init()) method, use the [Requirements API](/developer_guides/templates/requirements/#php-requirements-api) to add any JavaScript that your implementation needs to function in both the front-end and in the CMS. This should not include JavaScript that is only required in the CMS, which we'll cover in the next section.
+
+If your HTML editor's JavaScript implementation has its own client-side allow list of elements and attributes, you can use the [`getElementRuleSet()`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig::getElementRuleSet()) and [`setElementRuleSet()`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig::setElementRuleSet()) to convert to and from that client-side definition to allow developers to choose whether to use the generic element rulesets or the implementation-specific configuration. This can be usedful if, for example, your JavaScript implementation provides additional configuration options in its rule sets.
+
+Alternatively, you can convert from the ruleset in either `getAttributes()` or `getConfigSchemaData()`, and just set the element rule set as a property in your config.
+
+Use [`getOptions()`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig::getOptions()) and [`setOptions()`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorConfig::setOptions()) for generic options that can be defined for your implementation.
+
+If you want your editor to be the default for all `HTMLEditorConfig`, set it through the injector configuration:
+
+```yml
+SilverStripe\Core\Injector\Injector:
+  SilverStripe\Forms\HTMLEditor\HTMLEditorConfig:
+    class: 'App\Forms\CustomHTMLEditorConfig'
 ```
 
-## Developing a wrapper to use a different WYSIWYG editors with `HTMLEditorField`
+If you want your editor to be the default specifically for the `cms` config, set the `HTMLEditorConfig.default_config_definitions` configuration like so:
```

Comment:
This section is admittedly more for us than for anyone else, so we know what needs to be done when we find a suitable replacement for TinyMCE.